src/vmm/src/rpc_interface.rs

```diff
@@ -2052,6 +2074,27 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_runtime_hotplug_vcpu() {
+        let req = VmmAction::HotplugVcpu(HotplugVcpuConfig { vcpu_count: 4 });
+        check_runtime_request(req, |result, vmm| {
+            assert_eq!(result, Ok(VmmData::Empty));
+            assert!(vmm.hotplug_vcpus_called)
+        });
+
+        let req = VmmAction::HotplugVcpu(HotplugVcpuConfig { vcpu_count: 0 });
+        check_runtime_request_err(
+            req,
+            VmmActionError::HotplugVcpu(HotplugVcpuError::VcpuCountTooLow),
+        );
+
+        let req = VmmAction::HotplugVcpu(HotplugVcpuConfig { vcpu_count: 33 });
+        check_runtime_request_err(
+            req,
+            VmmActionError::HotplugVcpu(HotplugVcpuError::VcpuCountTooHigh),
+        );
+    }
+
```

Comment:
As we discussed offline, this test actually only tests the big `match` statement at the top of this file, but never actually calls your `vcpu_hotplug` method. That's a preexisting issue with the code, so obviously not something you'll need to (or should!) fix here. However, it means that we can probably remove this entire test (and any interaction you've written wit the `rpc_interface.rs` mocking stuff).

Instead, lets write a unit test in `lib.rs` that uses `default_vmm` and then calls the actual `Vmm::hotplug_vcpu`.

For posterity, because James and me were wondering for a while: How does this test even pass, if `VcpuCountTooLow` is only returned in one location, and that location is reported by codecov as uncovered? The answer is that the mocks consider two `VmmActionError`s to be equal if they are the same variant, without actually taking into consideration what the variant contains (so `VmmActionError::HotplugVcpu(HotPlugVcpuError::VcpuCountTooLow)` compares equal to `VmmActionError::HotplugVcpu(HotPlugVcpuError::VcpuCountTooHigh)`)